Core/Src/main.c

```diff
@@ -65,7 +71,8 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
-
+	char spi_buf_rx[10];
+	char spi_buf_tx[3] = {0, 0, 0}; // setting D2, D1, D0 to 0 for mode and channel
```

Comment:
The type of this buffer should be `uint8_t` which is the same size as `char`, but it is unsigned. The 'uint8_t' is also the required data type to pass into the `HAL_SPI_TransmitReceive` function

Reply:
Not correct tx data btw, I will probably teach you how to read datasheet